app/components/newsletter/newsletter-header-overview.js

```diff
@@ -1,154 +1,182 @@
+      this.isLoading = false;
+    });
+    this.newsletterHTML = null;
+    this.testCampaignIsLoading = false;
+    location.reload();
+  }
+
+  @action
+  async deleteCampaign() {
+    this.isLoading = true;
+    const agenda = await this.args.agenda;
+    const meeting = await agenda.createdFor;
+    const mailCampaign = await meeting.mailCampaign;
+    if (mailCampaign && mailCampaign.campaignId) {
+      await this.newsletterService.deleteCampaign(mailCampaign.campaignId);
+    }
+    mailCampaign.destroyRecord();
+    const reloadedMeeting = await this.store.findRecord('meeting', meeting.id, {
+      reload: true,
+    });
+    reloadedMeeting.mailCampaign = null;
+
+    this.mailCampaign = null;
+    await reloadedMeeting.save();
+    this.isLoading = false;
+    location.reload();
+  }
+
+  @action
+  async sendCampaign() {
+    this.isLoading = true;
+    const agenda = await this.args.agenda;
+    const meeting = await agenda.createdFor;
+    const mailCampaign = await meeting.mailCampaign;
+
+    if (!mailCampaign || !mailCampaign.id || mailCampaign.isSent) {
+      return;
+    }
+
+    let validCampaign = true;
+    await this.validatedCampaign(mailCampaign.campaignId).catch((ex) => {
+      console.error(ex);
+      this.isVerifying = false;
+      this.isLoading = false;
+      validCampaign = false;
+    });
+    if (!validCampaign) {
+      return false;
+    }
+
+    await this.newsletterService
+      .sendCampaign(mailCampaign.campaignId, agenda.id)
+      .catch(() => {
+        this.toaster.error(
+          this.intl.t('error-send-newsletter'),
+          this.intl.t('warning-title')
+        );
       });
-      this.set('isLoading', false);
-      mailCampaign.set('sentAt', moment().utc()
-        .toDate());
-      await mailCampaign.save();
-      await meeting.belongsTo('mailCampaign').reload();
-      this.set('isVerifying', false);
-    },
-
-    async sendTestCampaign() {
-      this.set('testCampaignIsLoading', true);
-      const agenda = await this.get('agenda');
-      const meeting = await agenda.get('createdFor');
-      const mailCampaign = await meeting.get('mailCampaign');
-
-      if (!mailCampaign || !mailCampaign.id || mailCampaign.isSent) {
-        this.set('newsletterHTML', html.body);
-        this.set('testCampaignIsLoading', false);
-        return;
-      }
-
-      const html = await this.newsletterService.getMailCampaignContent(mailCampaign.campaignId).catch(() => {
-        this.toaster.error(this.intl.t('error-send-newsletter'), this.intl.t('warning-title'));
+    this.isLoading = false;
+    mailCampaign.set('sentAt', moment().utc().toDate());
+    await mailCampaign.save();
+    await meeting.belongsTo('mailCampaign').reload();
+    this.isVerifying = false;
+  }
+
+  @action
+  async sendTestCampaign() {
+    this.testCampaignIsLoading = true;
+    const agenda = await this.args.agenda;
+    const meeting = await agenda.createdFor;
```

Comment:
op zich weer buiten scope van de octane refactor, maar zijn die awaits nodig als we zowel `agenda` als `meeting`via args binnenkrijgen

Reply:
Het waren geen grote aanpassingen dus ik heb ze meegenomen.